index.js

```diff
@@ -189,25 +189,47 @@ module.exports = class Hyperdrive extends ReadyResource {
     return this.files.del(normalizePath(name))
   }
 
-  async clear (name, { storageInfo = false } = {}) {
+  async clear (name, opts) {
+    const cleared = opts && opts.diff ? { blocks: 0 } : null
+
     const node = await this.entry(name)
-    if (node === null) return 0
+    if (node === null) return cleared
 
     await this.getBlobs()
 
-    const infoBefore = storageInfo
+    const infoBefore = cleared
       ? await this.blobs.core.info({ storage: true })
       : null
 
     await this.blobs.clear(node.value.blob)
 
-    if (infoBefore) {
-      const infoAfter = await this.blobs.core.info({ storage: true })
-      const bytesCleared = infoBefore.storage.blocks - infoAfter.storage.blocks
-      return bytesCleared
-    }
+    if (!infoBefore) return cleared
```

Comment:
`if (!cleared) return null` is clearer imo

Reply:
I think `core.info()` can return null or something like that, that's why the original code checks for `infoBefore`

Reply:
Also, we will use the returned value of `core.clear` so all this won't matter that much literally by tomorrow max haha

Reply:
if cleared is null the line below fails anyway. they are coupled

Reply:
fixed